### DISCUSSION TITLE
v2.2.2 (Beta)

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#3071] [Releases]
## Installation
Visit [https://getutm.app/install/][1] for the most up to date installation instructions.

## New Features (v2.2.0)
* (macOS 11.3+ Only) **Bridged networking and shared networking support**. You can configure it in the Network settings for your VM.
* (macOS 11+ and iOS 13+) **GPU acceleration for OpenGL on Linux**. Use `virtio-ramfb` or `virtio-gpu-gl-pci` display device and compatible Linux drivers (most modern Linux distros will have it already installed). Windows is not supported because there is currently no virtio-gpu driver for Windows that supports 3D acceleration.

## Changes (v2.2.2)
* (iOS) Enable OpenGL acceleration for iOS 11-12
* (iOS 15+) Increased memory limit for supported devices (M1 iPads currently)
* (macOS) Fix layout issues in settings on macOS Monterey (#2644)
* (macOS) Fix capturing of hotkeys (such as Cmd+Tab) (#2677)
* Disable port forwarding options for shared & bridged networking
* Fixed crash when PPC screamer is selected on iOS (currently this is still not supported)
* Intel Macs will use EFI boot if virtio-* graphics card is selected (fixes black screen on boot)
* Refactored non-OpenGL rendering code, fixing some random crashes seen in TestFlight reports.

Please check the full list on [Github][2] and help report any bug you find that is not listed.

[1]: https://getutm.app/install/
[2]: https://github.com/utmapp/UTM/issues

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v2.2.2'>v2.2.2 (Beta)</a>.</em>

**Comment by conath** [comment]
> Intel Macs will use EFI boot if virtio-* graphics card is selected (fixes black screen on boot)

this could cause some compatibility issues with existing VMs, no? Or does this only apply to new VMs?

Edit: no, this configuration was previously unsupported so it will not cause incompatibility.

**Reply by osy** [reply]
Currently if you are on x86 and you use virtio-* and you don't have a custom pflash EFI image, it won't boot. If you have a custom pflash specified then it will take precedence (same on ARM64).

**Reply by conath** [reply]
Was the virtio-card not booting with BIOS always the case? Reading the 2.2.0 release notes it sounded like that was a new issue.

**Reply by osy** [reply]
It never worked for me. I put it in the release notes because I know people will run it (considering it's the only way to enable GL) and see that it doesn't boot.

**Reply by conath** [reply]
Alright then I am relieved :)

**Reply by kode54** [reply]
The gpu-gl-device option is failing with a black screen, and gpu-gl-pci boots with the stock efi image on M1, but the firmware and boot loader's printing options are incredibly slow, as if each individual draw operation is waiting for a vsync cycle. And then it fails with some GL cookie error when the VM gets to actually trying to use graphics.

ramfb seems to work, though, but has incredibly slow virgl performance, as if it is a soft GPU.

**Reply by conath** [reply]
Please [create an issue](https://github.com/utmapp/UTM/issues/new/choose) and include the data requested in the issue template. Thank you!

**Comment by conath** [comment]
I have tested the following guests on my M1 MacBook Pro (macOS 11.5.2); system architecture is in braces:

- Mac OS 9 (PowerPC) OK
- DOS (i368 / x86) OK
- Windows 11 for ARM (aarch64 with Hypervisor; "Shared Network" mode; `virtio-gpu-gl-pci` with DOD drivers) OK
 => #3023 and #2628 persist, as expected; 2D accel working just as before.
- Ubuntu 20.04 LTS ARM OK: Desktop works, 3D crashes as expected #3053
- ReactOS (x86_64) OK
- Windows 10 21h1 (x86_64): **FAIL**: existing VM using virtio-gpu-pci booted fine once, but froze and restarted. Now boot loop (seems to restart after Windows boot/when enabling graphics mode). Changing to `qxl-vga` now does not help! 
[debug.log](https://github.com/utmapp/UTM/files/7086497/debug.log)
  While trying to boot the installer I get #2721.

**Reply by tomkoi** [reply]
Finally i able to get network outside from the VM!
thanks!!!!

**Reply by tomkoi** [reply]
tried now install windows10 after download arm image from Microsoft - first time work, then the mouse went crazy as described. download again fresh file and convert him to qcow2 -- until now i not able to load again the image to UTM  

EDIT: seems i forgot to change the main drive (qcow2) to NVME, now seems better

**Reply by diegov12** [reply]
Whenever I try to start a VM in Monterey, I get an error message in Chinese. Tried to start ARM windows 10.
Translated, An error occurred while starting QEMU..

**Comment by vilgus** [comment]
M1 MacBook Air (macOS 11.5) - can not install Windows 7. I get BSOD (a memory page error) during the installation several minutes after the installation starts. If I run existing Windows 7 VMs created under UTM 2.1.2, they start normally but crash about the same time (in 2-3 mins). I had Windows 7 network issues in 2.1.2 and they seem to be fixed in 2.2.2. But now the whole operating system crashes. Please fix.

**Reply by conath** [reply]
Thanks for your detailed message. We are [aware](https://github.com/utmapp/UTM/issues/2721) of a recently introduced Windows BSOD issue with the beta and have [reverted](https://github.com/utmapp/UTM/commit/634ddaaf6308994fed71b7a4a7fd15840961fbd2) a backend update (QEMU) to mitigate this. There will be another beta release with the fix soon, and you will see it on the [Releases](https://github.com/utmapp/UTM/releases) page.

**Reply by conath** [reply]
The beta update that includes the bugfix is now available. [2.2.3](https://github.com/utmapp/UTM/releases/tag/v2.2.3)
